app/controllers/questions_controller.rb

```diff
@@ -2,67 +2,65 @@ class QuestionsController < ApplicationController
 
   before_action :set_question, only: [:show, :edit, :update, :destroy]
 
-      def index
-        @questions = Question.all
-      end
+  def index
+    @questions = Question.all
+  end
 
-      def show
-        @quiz = Quiz.find(params[:quiz_id])
-        @question = Question.find(params[:id])
-      end
+  def show
     
-      def new
-        @question = Question.new 
-      end
+  end
 
-    
-      def edit
-        @quiz = Quiz.find(params[:quiz_id])
-        @question = Question.find(params[:id])
-      end
-    
-      def update
-        if (@question.update(question_params))
-          redirect_to @quiz
-        else
-          render :edit
-         end
-      end
+  def new
+    @question = Question.new 
+  end
 
+  def edit
     
-      def create      
-        @quiz = Quiz.find(params[:quiz_id])
-        @question = Question.new(question_params)
-        correct_index = params[:question][:options_attributes][:correct_key]
-        if correct_index.present?
-          update_correct_answer = @question.options[correct_index.to_i]
-          update_correct_answer.update(:correct_key => true)
-          if @question.save
-            redirect_to @quiz
-          else
-            flash[:error] = "Fill all questions and options"
-            redirect_to @quiz
-          end
-        else  
-            flash[:error] = "Fill all questions, options and select correct key"
-            redirect_to @quiz
-        end 
-      end
+  end
 
-      def destroy
-        @question.destroy
-        redirect_to quiz_path(@quiz)
-      end
-      
-      private
+  def update
+    if (@question.update(question_params))
+      redirect_to @quiz
+    else
+      render :edit
+    end
+  end
 
-      def set_question
-        @quiz = Quiz.find(params[:quiz_id])
-        @question = Question.find(params[:id])
+  def create      
+    @quiz = Quiz.find(params[:quiz_id])
+    @question = Question.new(question_params)
+    correct_index = params[:question][:options_attributes][:boolean_correct_answer]
+    if correct_index.present?
+      update_correct_answer = @question.options[correct_index.to_i]
```

Comment:
this can be modified to correct_answer instead of update_correct_answer right? just a suggestion to make it simple.